include/caffe/util/gpu_memory.hpp

```diff
@@ -0,0 +1,121 @@
+#ifndef CAFFE_UTIL_GPU_MEMORY_HPP_
+#define CAFFE_UTIL_GPU_MEMORY_HPP_
+
+#include <vector>
+#include "caffe/common.hpp"
+
+namespace caffe {
+
+class gpu_memory {
```

Comment:
Classes are `CamelCase` in Google style/Caffe. This is also easily confused with `SyncedMem`...